[RHCLOUD-20860] Refactor set OrgIds and EBS account numbers from the database

MikelAlejoBR opened the pull request:
Instead of just forwarding the incoming headers, this change ensures
that the "EBS account number" and the "OrgId" are set in the context
using the contents from the database.

This comes from one issue we have been having with the notifications
service: the service was erroring because we were just sending the
incoming "EBS account number" header, even though we had the associated
"OrgId" on the database.

So we have decided to simply use the incoming headers as a validation
—make sure that an EBS account number and/or OrgId are present—, and
then use the database contents to send the headers to the rest of the
services or to the Kafka queues.

## Links

[[RHCLOUD-20860]](https://issues.redhat.com/browse/RHCLOUD-20860)

MikelAlejoBR committed:
refactor: return the entire tenant when creating/finding it

This will help storing the EBS account number and OrgId numbers that we
have on the database in the context.

MikelAlejoBR committed:
refactor: tenancy middleware sets the tenant data from the database

Instead of just forwarding the incoming headers, this change ensures
that the "EBS account number" and the "OrgId" are set in the context
using the contents from the database.

This comes from one issue we have been having with the notifications
service: the service was erroring because we were just sending the
incoming "EBS account number" header, even though we had the associated
"OrgId" on the database.

So we have decided to simply use the incoming headers as a validation
—make sure that an EBS account number and/or OrgId are present—, and
then use the database contents to send the headers to the rest of the
services or to the Kafka queues.

lindgrenj6 reviewed (approved):
Barring the loggerfields part looks good!

MikelAlejoBR committed:
feature: update the identity struct with the tenancy data from the db

That way the entire application can benefit from the data we hold at the
database, instead of relying on just the incoming headers' data.

MikelAlejoBR committed:
refactor: rearrange the logger middleware so proper log fields are set

If the logger middleware is executed before the tenancy middleware, the
OrgId and EBS account numbers might not be the ones that we have from
the database, which could lead to incomplete logs.

lpichler reviewed (approved):
LGTM This is works great! Thanks again!
